### ISSUE TITLE
Update `userCanCreatePages` to `canUserCreatePages` for consistency

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#35253]
IN https://github.com/WordPress/gutenberg/pull/35083#discussion_r719187544_ @kevin940726 suggested we should rename the `userCanCreatePages` variable to `canUserCreatePages`. This is utilised in both:

* Nav block
* Inline within the format UI.

...to determine whether a user can create pages.

We should update it for consistency.

----------

I agree other than the fact that the Nav block already uses the `userCanCreatePages` pattern. Happy to raise a follow up to update this though as I agree your suggestion might be preferable.

_Originally posted by @getdave in https://github.com/WordPress/gutenberg/pull/35083#discussion_r719187544_

**Comment by im3dabasia** [comment]
Hey @getdave , This issue seems to be there in the Gutenberg codebase. Its a minor one though. Are we still looking forward for this change? If yes I can take it up.

cc: @t-hamano

**Comment by t-hamano** [comment]
If we were to rename it, we should consider the following two points at the same time:

- Whether to stabilize this property (from `__experimentalUserCanCreatePages` to `canUserCreatePages`)
- Whether to maintain backward compatibility

The question here is what kind of backward compatibility policy to apply. In my opinion, this property is very niche, so I think it would be enough to simply rename it to `canUserCreatePages` and mention it in the dev note.

cc @WordPress/gutenberg-core

**Comment by Mamaduka** [comment]
If we don't "care enough" about these experimental settings to maintain backward compatibility, perhaps we should leave them as experimental.

I think the only reason they're exposed as `block-editor` settings is because of the Link format, where we wanted to avoid introducing the `core-data` package as a dependency.

**Comment by t-hamano** [comment]
Yes, I think it's also fine to leave them experimental and close this issue as "won't fix".
